src/libs/ERC712.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.23;
+
+import { SignatureChecker } from "./SignatureChecker.sol";
+
+/// @title Typed structured data hashing and signing via EIP-712.
+/// @dev   An abstract implementation to satisfy EIP-712: https://eips.ethereum.org/EIPS/eip-712
+library ERC712 {
+    // keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)")
+    bytes32 internal constant _EIP712_DOMAIN_HASH = 0x8b73c3c69bb8fe3d512ecc4cf759cc79239f7b179b0ffacaa9a75d522b39400f;
+
+    // keccak256("1")
+    bytes32 internal constant _EIP712_VERSION_HASH = 0xc89efdaa54c0f20c7adf612882df0950f5a951637e0307cdcb4c672f298b8bc6;
+
+    /// @notice Revert message when an invalid signature is detected.
+    error InvalidSignature();
+
+    /// @notice Revert message when a signature with invalid length is detected.
+    error InvalidSignatureLength();
+
+    /**
+     * @notice Revert message when a signature is being used beyond its deadline (i.e. expiry).
+     * @param  deadline  The deadline of the signature.
+     * @param  timestamp The current timestamp.
+     */
+    error SignatureExpired(uint256 deadline, uint256 timestamp);
+
+    /// @notice Revert message when a recovered signer does not match the account being purported to have signed.
+    error SignerMismatch();
+
+    /**
+     * @notice Computes the EIP-712 domain separator.
+     * @param  name_ The name of the contract
+     * @return       The EIP-712 domain separator
+     */
```

Comment:
```suggestion
    /**
     * @notice Computes the EIP-712 domain separator.
     * @param  name_ The name of the contract
     * @return The EIP-712 domain separator
     */
```

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/7/commits/a27206542da32d6caa18a62cbfcbf772bf11a332